src/app.js

```diff
@@ -3,36 +3,46 @@ import cors from 'cors'
 import helmet from 'helmet'
 import Honeybadger from '@honeybadger-io/js'
 import jwt from 'express-jwt'
+import * as OpenApiValidator from 'express-openapi-validator'
 import connect from './mongo.js'
 import jwtConfig from './jwt.js'
 import resourcesRouter from './endpoints/resources.js'
 import groupsRouter from './endpoints/groups.js'
 import marcRouter from './endpoints/marc.js'
 import usersRouter from './endpoints/users.js'
-import { errorHandler, mongoErrorAdapter, s3ErrorAdapter } from './error.js'
+import { errorHandler, mongoErrorAdapter, s3ErrorAdapter, openApiValidatorErrorHandler, jwtErrorAdapter } from './error.js'
 
-const app = express()
+const isProduction = process.env.NODE_ENV === 'production'
 
-// Use HB before all other app middleware.
 Honeybadger.configure({
   apiKey: process.env.HONEYBADGER_API_KEY,
   environment: process.env.HONEYBADGER_ENV
 })
 
-app.use(function (req, res, next) {
-  if(process.env.NODE_ENV === 'development') console.log(`${req.method} ${req.url}`)
-  next()
-})
-
-// Increase the allowed payload size.
-app.use(express.json({limit: '1mb'}))
+const app = express()
 
-// CORS should probably be tightened down.
+// Handle CORS before openapi validation.
 app.use(cors({exposedHeaders: [
   'Location',
   'Content-Location'
 ]}))
 app.options('*', cors())
+
+// Increase the allowed payload size.
+// Required before OpenApiValidator.
+app.use(express.json({limit: '1mb'}))
+
+app.use(OpenApiValidator.middleware({
```

Comment:
This enables openapi validation for requests (always) and responses (when not in production).